src/chain/chain-generic-numerator.h

```diff
@@ -0,0 +1,173 @@
+// chain/chain-generic-numerator.h
+
+// Copyright       2017  Hossein Hadian
+
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef KALDI_CHAIN_CHAIN_GENERIC_NUMERATOR_H_
+#define KALDI_CHAIN_CHAIN_GENERIC_NUMERATOR_H_
+
+#include <vector>
+#include <map>
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "fstext/fstext-lib.h"
+#include "tree/context-dep.h"
+#include "lat/kaldi-lattice.h"
+#include "matrix/kaldi-matrix.h"
+#include "hmm/transition-model.h"
+#include "chain/chain-supervision.h"
+#include "cudamatrix/cu-matrix.h"
```

Comment:
I think you are including files that are not required or can be included in cc files.

Reply:
Yes, I guess these were for some unusual experiments.